src/main/scala/RewriteView.scala

```diff
@@ -0,0 +1,109 @@
+package fuselang
+
+/**
+ * AST pass to rewrite views into simple array accesses. Should be used after
+ * type checking.
+ *
+ * For `shrink` views, we rewrite them as:
+ *
+ * view v_a = shrink a[4 * i : 4];
+ * v_a[k];
+ * ==>
+ * ; // remove the view decl
+ * a[4*i + k]
+ *
+ * If `a` itself is a view, we keep rewriting it until we reach a true array.
+ */
+object RewriteView {
+  import Syntax._
+  import CodeGenHelpers._
+
+	// We can use a flat environment because the type checker makes sure things
```

Comment:
Errant indentation?

Reply:
OMG it was an actual TAB. I've never seen one before. Idk how it happened.